students/alsabhan/Alsabhan_econ5316_HW_7_&_8.rmd

```diff
@@ -0,0 +1,261 @@
+
+
+
+
+
+#HW8
+```{r}
+
+COP.Data <- tq_get("MCOILWTICO", get = "economic.data", from = "1995-01-01", to = "2017-12-31")
+COP_tbl <-
+  COP.Data %>%
+  as_tibble() %>%
+  mutate(yearq = as.yearqtr(date, format = "%Y:%q"))%>%
+  dplyr::select(c(yearq,price))
+
+COP_ts <-
+  COP_tbl %>%
+  tk_ts(select = -yearq, start = year(.$yearq[1]), frequency = 4)
+log.COP <- log(COP_ts)
+f.diff.COP <- diff( log(COP_ts), differences = 1)
+
+
+RCGP.Data <- tq_get("GASREGCOVM", get = "economic.data", from = "1995-01-01", to = "2017-12-31")  
+RCGP_tbl<-
+  RCGP.Data %>%
+  as_tibble() %>%
+  mutate(yearq = as.yearqtr(date, format = "%Y:%q"))%>%
+  dplyr::select(c(yearq,price))
+
+RCGP_ts <-
+  RCGP_tbl %>%
+  tk_ts(select = -yearq, start = year(.$yearq[1]), frequency = 4)
+log.RCGP <- log(RCGP_ts)
+f.diff.RCGP <- diff(log(RCGP_ts), differences = 1)
+
+par(mar=c(1,1,1,1))
+plot(log.COP, type='l', main="The Log of Oil and Gas Prices from 1995 to 2017"
+     ,col="blue", ylim=c(-2, 5),
+     xlab= "Year", ylab="log of Price", colour='variable' )
+lines(log.RCGP, col="red")
+
+plot(f.diff.COP, type='l', main="Difference of the log of Oil and log of Gas from 1955 to 2017", col="blue",
+     ylim=c(-0.4, 0.4))
+lines(f.diff.RCGP, col="red")
+```
+```{r}
+#b
+log.COP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                   lag.max = 8) %>%
+  summary()
+f.diff.COP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                      lag.max = 8) %>%
+  summary()
+
+log.COP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                   lag.max = 8) %>%
+  summary()
+f.diff.COP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                      lag.max = 8) %>%
+  summary()
+
+log.RCGP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                   lag.max = 8) %>%
+  summary()
+f.diff.RCGP %>% ur.ers( type = c("DF-GLS", "P-test"), model = c( "trend"),
+                      lag.max = 8) %>%
+  summary()
+```
+
+```{r}
+#c
+c <- cbind(f.diff.COP,f.diff.RCGP)
+colnames(c) <- c("log.oil","log.gas")
+c <- na.trim(c)
+VAR.p <- VARselect(c, type="const")
+numlags <- VAR.p$selection["SC(n)"]
+numlags
+
+c <- window(c, start=1995+0, end=2017+12/12)
+c.1 <- ca.jo(c, ecdet="const", type="trace", K=numlags, spec="transitory")
+summary(c.1)
+c.2 <- ca.jo(c, ecdet="const", type="eigen", K=numlags, spec="transitory")
+summary(c.2)
+
+lttest(c.1, r=1)
+```
+
+```{r}
+#d
+restricted <- matrix(c(1,0), c(2,1))
+C1.restricted <- alrtest(c.2, A = restricted, r = 1)
+VEC.restricted <- cajorls(C1.restricted, r = 1)
+summary(VEC.restricted)
+```
+
+```{r}
+#e
+VEC <- cajorls(c.1, r=1)
+VEC
+```
+
+```{r}
+#f
+summary(VEC$rlm)
+```
+
+# α1 is not statistically significant while α2 is statistically significant. Therefore,there is distrubtion in the long 
+#run equilibrium 
+```{r}
+#g
+rest.alpha <- matrix(c(1,0), c(2,1))

```

Comment:
since your variables are in the opposite order thenspecified in HW8 setup, 
see the line y <- cbind(f.diff.RCGP, f.diff.COP) above,
here you need to use
rest.alpha <- matrix(c(0,1), c(2,1))
in order to impose restriction alpha1 = 0 because alpha1 is not statistically significant